server/datastore/mysql/mdm.go

```diff
@@ -314,3 +315,306 @@ WHERE
 		return nil
 	})
 }
+
+func (ds *Datastore) UpdateHostMDMProfilesVerification(ctx context.Context, host *fleet.Host, toVerify, toFail, toRetry []string) error {
+	return ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		if err := setMDMProfilesVerifiedDB(ctx, tx, host, toVerify); err != nil {
+			return err
+		}
+		if err := setMDMProfilesFailedDB(ctx, tx, host, toFail); err != nil {
+			return err
+		}
+		if err := setMDMProfilesRetryDB(ctx, tx, host, toRetry); err != nil {
+			return err
+		}
+		return nil
+	})
+}
+
+// setMDMProfilesRetryDB sets the status of the given identifiers to retry (nil) and increments the retry count
+func setMDMProfilesRetryDB(ctx context.Context, tx sqlx.ExtContext, host *fleet.Host, identifiersOrNames []string) error {
```

Comment:
heads-up: all the new functions in this file were mostly moved from `server/datastore/mysql/apple_mdm.go` but they contain modifications to support both platforms.